File: gyswp-admin/src/entities/document.ts
export class DocumentDto {
    id: string;
    name: string;
    docNo: string;
    categoryId: number;
    categoryDesc: string;
    docThum: string;
    summary: string;
    publishTime: string;
    qrCodeUrl: string;
    deptIds: string;
    deptDesc: string;
    isDeleted: boolean;
    creationTime: Date;
    creatorUserId: string;
    lastModificationTime: Date;
    lastModifierUserId: string;
    deletionTime: Date;
    deleterUserId: string;
    employeeIds: string;
    employeeDes: string;
    isAllUser: boolean;
    deptName: string;
    isAction: boolean;
    stamps: string;
    suitableCode: string;
    bllId: string;
    constructor(data?: any) {
        if (data) {
            for (var property in data) {
                if (data.hasOwnProperty(property))
                    (<any>this)[property] = (<any>data)[property];
            }
        }
    }

    getUsers(): any[] {
        let users = [];
        if (!this.employeeIds) {
            this.employeeIds = '';
        }
        if (!this.employeeDes) {
            this.employeeDes = '';
        }
        let ids = this.employeeIds.split(',');
        let names = this.employeeDes.split(',');
        let i = 0;
        for (let id of ids) {
            if (id) {
                users.push({ id: id, name: names[i] });
            }
            i++;
        }
        return users;
    }

    getDepts(): any[] {
        let depts = [];
        if (!this.deptIds) {
            this.deptIds = '';
        }
        if (!this.deptDesc) {
            this.deptDesc = '';
        }
        let ids = this.deptIds.split(',');
        let names = this.deptDesc.split(',');
        let i = 0;
        for (let id of ids) {
            if (id) {
                depts.push({ id: id, name: names[i] });
            }
            i++;
        }
        return depts;
    }

    init(data?: any) {
        if (data) {
            this.id = data["id"];
            this.name = data["name"];
            this.docNo = data["docNo"];
            this.categoryId = data["categoryId"];
            this.categoryDesc = data["categoryDesc"];
            this.docThum = data["docThum"];
            this.summary = data["summary"];
            this.publishTime = data["publishTime"];
            this.qrCodeUrl = data["qrCodeUrl"];
            this.deptIds = data["deptIds"];
            this.deptDesc = data["deptDesc"];
            this.isDeleted = data["isDeleted"];
            this.creationTime = data["creationTime"];
            this.creatorUserId = data["creatorUserId"];
            this.lastModificationTime = data["lastModificationTime"];
            this.lastModifierUserId = data["lastModifierUserId"];
            this.deletionTime = data["deletionTime"];
            this.deleterUserId = data["deleterUserId"];
            this.employeeIds = data["employeeIds"];
            this.employeeDes = data["employeeDes"];
            this.isAllUser = data["isAllUser"];
            this.deptName = data["deptName"];
            this.isAction = data["isAction"];
            this.stamps = data["stamps"];
            this.suitableCode = data["suitableCode"];
            this.bllId = data["bllId"];
        }
    }
    toJSON(data?: any) {
        data = typeof data === 'object' ? data : {};
        data["id"] = this.id;
        data["name"] = this.name;
        data["docNo"] = this.docNo;
        data["categoryId"] = this.categoryId;
        data["categoryDesc"] = this.categoryDesc;
        data["docThum"] = this.docThum;
        data["summary"] = this.summary;
        data["publishTime"] = this.publishTime;
        data["qrCodeUrl"] = this.qrCodeUrl;
        data["deptIds"] = this.deptIds;
        data["deptDesc"] = this.deptDesc;
        data["isDeleted"] = this.isDeleted;
        data["creationTime"] = this.creationTime;
        data["creatorUserId"] = this.creatorUserId;
        data["lastModificationTime"] = this.lastModificationTime;
        data["lastModifierUserId"] = this.lastModifierUserId;
        data["deletionTime"] = this.deletionTime;
        data["deleterUserId"] = this.deleterUserId;
        data["employeeIds"] = this.employeeIds;
        data["employeeDes"] = this.employeeDes;
        data["isAllUser"] = this.isAllUser;
        data["isAction"] = this.isAction;
        data["stamps"] = this.stamps;
        data["suitableCode"] = this.suitableCode;
        data["bllId"] = this.bllId;
        return data;
    }
    static fromJS(data: any): DocumentDto {
        let result = new DocumentDto();
        result.init(data);
        return result;
    }
    static fromJSArray(dataArray: any[]): DocumentDto[] {
        let array = [];
        dataArray.forEach(result => {
            let item = new DocumentDto();
            item.init(result);
            array.push(item);
        });
        return array;
    }
    clone() {
        const json = this.toJSON();
        let result = new DocumentDto();
        result.init(json);
        return result;
    }
}

File: aspnet-core/aspnet-core/src/GYSWP.Core/Documents/Document.cs
using Abp.Domain.Entities.Auditing;
using GYSWP.GYEnums;
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace GYSWP.Documents
{
    [Table("Documents")]
    public class Document : FullAuditedEntity<Guid> //注意修改主键Id数据类型
    {
        /// <summary>
        /// 标准名称
        /// </summary>
        [StringLength(200)]
        [Required]
        public virtual string Name { get; set; }
        /// <summary>
        /// 标准编号
        /// </summary>
        [StringLength(100)]
        //[Required]
        public virtual string DocNo { get; set; }
        /// <summary>
        /// 标准分类Id
        /// </summary>
        [Required]
        public virtual int CategoryId { get; set; }
        /// <summary>
        /// 分类Id层级 用逗号分隔
        /// </summary>
        [StringLength(200)]
        public virtual string CategoryCode { get; set; }
        /// <summary>
        /// DeptIds
        /// </summary>
        public virtual string DeptIds { get; set; }
        /// <summary>
        /// DeptDesc
        /// </summary>
        public virtual string DeptDesc { get; set; }
        /// <summary>
        /// 分类名描述（分类名层级以逗号分隔）
        /// </summary>
        [StringLength(500)]
        public virtual string CategoryDesc { get; set; }
        /// <summary>
        /// 摘要说明
        /// </summary>
        [StringLength(2000)]
        public virtual string Summary { get; set; }
        /// <summary>
        /// 发布日期
        /// </summary>
        public virtual DateTime? PublishTime { get; set; }

        /// <summary>
        /// 授权员工名称（以逗号分隔）
        /// </summary>
        public virtual string EmployeeDes { get; set; }

        /// <summary>
        /// 是否是全部用户
        /// </summary>
        public virtual bool IsAllUser { get; set; }
        /// <summary>
        /// 员工授权Ids（以逗号分隔）
        /// </summary>
        public virtual string EmployeeIds { get; set; }

        /// <summary>
        /// 是否启用（作废）
        /// </summary>
        [Required]
        public virtual bool IsAction { get; set; }

        /// <summary>
        /// 作废时间
        /// </summary>
        public virtual DateTime? InvalidTime { get; set; }

        /// <summary>
        /// 电子章数组 （以逗号分隔）
        /// </summary>
        [StringLength(50)]
        public virtual string Stamps { get; set; }

        /// <summary>
        /// 适用于（QMS,EMS,OHS）
        /// </summary>
        public virtual string SuitableCode { get; set; }
        /// <summary>
        /// 修订记录表Id
        /// </summary>
        public virtual Guid BLLId { get; set; }
    }
}
